container_ci_suite/engines/s2i_container.py

```diff
@@ -464,49 +465,48 @@ def doc_content_old(self, strings: List) -> bool:
     # }
 
     def test_response(
-            self, url: str = "",
-            expected_code: int = 200, port: int = 8080,
-            expected_output: str = "", max_tests: int = 20
+            self, url: str = "", port: int = 8080,
+            expected_code: int = 200, expected_output: str = "",
+            max_tests: int = 20
     ) -> bool:
-        url = f"{url}:{port}"
+        """
+        Test HTTP response of specified url
+
+        If expected output is empty, check will be skipped
+
+        Args:
+            url (str): URL where to send request
+            port (int): Port of the web service
+            expected_code (int): Check that response has this code
+            expected_output (str): Check that response has this text
+            max_tests (int): Stop after this many unsuccessful tries
+        """
+
+        url = f"http://{url}:{port}"
         print(f"URL address to get response from container: {url}")
-        cmd_to_run = "curl --connect-timeout 10 -k -s -w '%{http_code}' " + f"{url}"
-        # Check if application returns proper HTTP_CODE
-        print("Check if HTTP_CODE is valid.")
-        for count in range(max_tests):
+
+        try:
```

Comment:
This is not needed.

Before 'while' set response to None and let's go to while cycle.
Also return value is boolean.

The code 
```
(response is not None and
               response.status_code != expected_code and
               (response.text != expected_output or expected_output == "") and
               max_tests > 0)
```
is duplicated. In case all conditions are met, then return False.
Let's make the condition not onliner.

Reply:
Conditions were complicated, so I used simpler if statements to control the while loop.

Try/catch is needed because when request timeouts, it throws an exception. If we don't catch it, we fail the test immediately.